nilearn/reporting/sphinx_report.py

```diff
@@ -0,0 +1,34 @@
+# Scraper for sphinx-gallery
+# Copied from https://github.com/mne-tools/mne-python/
+
+import os.path as op
+from shutil import copyfile
+from nilearn.reporting import HTMLReport
+
+
+_SCRAPER_TEXT = '''
+.. only:: builder_html
+    .. container:: row
+        .. raw:: html
+            <iframe class="sg_report" sandbox="allow-scripts" src="{0}"></iframe>
+'''  # noqa: E501
+
+
+class _ReportScraper(object):
+    """Scrape Reports for Sphinx display.
+    """
+
+    def __init__(self):
+        self.app = None
+        self.files = dict()
+
+    def __repr__(self):
+        return '<ReportScraper>'
+
+    def __call__(self, block, block_vars, gallery_conf):
+        for report in block_vars['example_globals'].values():
+            if (isinstance(report, HTMLReport) and
+                    gallery_conf['builder_name'] == 'html'):
+                # embed links/iframe
+                data = _SCRAPER_TEXT.format(report.get_standalone())
+                return data
```

Comment:
It's missing the critical `return ''` at the end, this is what SG complains about (it must always return a `str`)

https://github.com/mne-tools/mne-python/blob/master/mne/report.py#L2120

Reply:
```suggestion
                return data
        return ''
```